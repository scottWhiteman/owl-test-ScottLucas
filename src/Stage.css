#Stage {
    background-color: #353535;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
}

#Stage .Participant {
    display: flex;
    flex-direction: row-reverse;
}

#Stage img {
    height: 50%;
}