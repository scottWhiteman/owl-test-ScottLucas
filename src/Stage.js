import React from 'react';
import Participant from './Participant';
import './Stage.css';

class Stage extends React.Component {
    static defaultProps = {
        participants: []
    };

    render() {
        const {participants} = this.props
        return (
            <div id="Stage">
                {participants.map(participant => (
                    <Participant 
                        id={participant.id}
                        avatar={participant.avatar}
                        name={participant.name}
                        onStage={participant.onStage}
                    />
                ))}
            </div>
        );
    }
}

export default Stage;