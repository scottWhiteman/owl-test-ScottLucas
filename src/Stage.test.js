import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Stage from './Stage';

it ("Participant List render without crash", () => {
    const div = document.createElement("div");
    ReactDOM.render(
        <Stage />
    , div);
    ReactDOM.unmountComponentAtNode(div);
});

it ("Participants render without crash", () => {
    const tree = renderer.create(<Stage />).toJSON();
    expect(tree).toMatchSnapshot();
});
