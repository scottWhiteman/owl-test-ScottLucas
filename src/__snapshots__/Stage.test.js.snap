// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participants render without crash 1`] = `
<div
  id="Stage"
/>
`;
